### PR TITLE
Allow `label`, `helpText`, `prefix` and `suffix` as params

### PR DESCRIPTION
**Pull request by tarwin**
Allow `label`, `helpText`, `prefix` and `suffix` as params on components that have them as slots. The slots override the params.

### CODE REVIEW

File: src/components/RangeSlider/RangeSlider.vue [hunk 1]
```diff
@@ -10,12 +10,15 @@ DualThumb(
   template(#label, v-if="slots.label || label")
     slot(v-if="slots.label", name="label")
     template(v-else) {{ label }}
-  template(#help-text, v-if="slots['help-text']")
-    slot(name="help-text")
-  template(#prefix, v-if="slots.prefix")
-    slot(name="prefix")
-  template(#suffix, v-if="slots.suffix")
-    slot(name="suffix")
+  template(#help-text, v-if="slots['help-text']" || helpText)
```
Comment (juzser) [inline src/components/RangeSlider/RangeSlider.vue:13]: Mistake

File: src/components/Checkbox/Checkbox.vue [hunk 2]
```diff
@@ -7,10 +7,12 @@ Choice(
   @mouseover="mouseOver = true",
   @mouseout="mouseOver = false",
 )
-  template(#label, v-if="slots.label")
-    slot(name="label")
-  template(#help-text, v-if="slots['help-text']")
-    slot(name="help-text")
+  template(#label, v-if="slots.label || label")
+    slot(name="label" v-if="slots.label")
+    template(v-else-if="label") {{ label }}
```
Comment (juzser) [inline src/components/Checkbox/Checkbox.vue:12]: I think we just need `v-else` here

File: src/components/TextField/TextField.vue [hunk 3]
```diff
@@ -78,6 +85,8 @@ Labelled(
         v-else,
         :is="multiline ? 'textarea' : 'input'",
         :id="id",
+        :label="label",
```
Comment (juzser) [inline src/components/TextField/TextField.vue:88]: Nah, why do we need this prop here?

**Review by juzser — COMMENTED** [review]
I cannot approve this PR, it should be improved a bit. 
But I think it's very close. If you need, I can update these features from my side.

**Comment by tarwin** [comment]
It's likely that I'm screwing up because I still don't quite understand pug - the rules coming over from HTML are a little ... different? Or I just don't read it as well. Yes, please make those small changes if you would, sorry.

**Comment by juzser** [comment]
No worry, you did a good job, just small tweaks needed. I'm updating this feature together with migrating Polaris v9.20.0.

**Comment by tarwin** [comment]
I know it's hard writing an open source project. You want people to be involved so you don't have to do it all yourself but you also want very specific standards :D

**Comment by juzser** [comment]
#155 solved this

**Pull request closed without merging**